packages/components/src/form-file-upload/test/index.js

```diff
@@ -1,30 +1,91 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
-import { noop } from 'lodash';
+import { render as RTLrender, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
  */
 import FormFileUpload from '../';
 
+/**
+ * Browser dependencies
+ */
+const { File } = window;
+
+function render( jsx ) {
+	return {
+		user: userEvent.setup( {
+			// Avoids timeout errors (https://github.com/testing-library/user-event/issues/565#issuecomment-1064579531).
+			delay: null,
+		} ),
+		...RTLrender( jsx ),
+	};
+}
+
+// @testing-library/user-event considers changing <input type="file"> to a string as a change, but it do not occur on real browsers, so the comparisons will be against this result
+const fakePath = expect.objectContaining( {
+	target: expect.objectContaining( {
+		value: 'C:\\fakepath\\hello.png',
+	} ),
+} );
+
 describe( 'FormFileUpload', () => {
 	it( 'should show an Icon Button and a hidden input', () => {
-		const wrapper = shallow(
+		render( <FormFileUpload>My Upload Button</FormFileUpload> );
+
+		const button = screen.getByText( 'My Upload Button' );
+		const input = screen.getByTestId( 'form-file-upload-input' );
+		expect( button ).toBeInTheDocument();
+		expect( input.style.display ).toBe( 'none' );
+	} );
+
+	it( 'should not fire a change event after selecting the same file', async () => {
+		const onChange = jest.fn();
+
+		const { user } = render(
+			<FormFileUpload onChange={ onChange }>
+				My Upload Button
+			</FormFileUpload>
+		);
+
+		const file = new File( [ 'hello' ], 'hello.png', {
+			type: 'image/png',
+		} );
+
+		const input = screen.getByTestId( 'form-file-upload-input' );
+
+		await user.upload( input, file );
+
+		await user.upload( input, file );
+
+		expect( onChange ).toHaveBeenNthCalledWith( 1, fakePath );
```

Comment:
Probably more correct?

```suggestion
		expect( onChange ).toHaveBeenCalledTimes( 1 );
```

Reply:
If we want to also check the arguments passed to `onChange`, it looks like we could also use [`toHaveBeenCalledWith()`](https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-)

Reply:
I didn't understand, [toHaveBeenNthCalledWith()](https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-) would be like these two functions in one, do I need to separate it in `toHaveBeenCalledTimes()` as @mirka said and `toHaveBeenCalledWith()` as @ciampo said or just left the way I implemented?

Reply:
I think it needs to _at least_ have a `toHaveBeenCalledTimes( 1 )` assertion for the test to be correct, because `toHaveBeenNthCalledWith( 1, foo )` does not check whether it was called more than once total. I think what @ciampo meant is that you could _also_ use `toHaveBeenCalledWith()` (or `toHaveBeenNthCalledWith()`) if you wanted to check the argument as well.

```js
// Necessary assertion for the test to be correct
expect( onChange ).toHaveBeenCalledTimes( 1 );

// Additional assertions that are largely interchangeable in this case
expect( onChange ).toHaveBeenNthCalledWith( 1, fakePath );
expect( onChange ).toHaveBeenCalledWith( fakePath );
```

Reply:
Exactly what @mirka said! I'm sorry if my previous comment wasn't as clear

Reply:
Fixed in 92a4a4d.